src/pages/TermsOfService.js

```diff
@@ -19,9 +19,11 @@ import * as Utils from 'src/libs/utils'
 const TermsOfServicePage = () => {
   const [busy, setBusy] = useState()
   const { isSignedIn, termsOfService } = authStore.get() // can't change while viewing this without causing it to unmount, so doesn't need to subscribe
-  const needToAccept = isSignedIn && !termsOfService.userAcceptedTos
-  const isGracePeriod = isSignedIn && termsOfService.isGracePeriodEnabled && !_.isUndefined(termsOfService.userAcceptedVersion)
+  const needToAccept = isSignedIn && termsOfService.showTosPopup
+  const userHasAcceptedPreviousTos = !_.isUndefined(termsOfService.userAcceptedVersion)
+  const canUserContinueUnderGracePeriod = isSignedIn && termsOfService.isGracePeriodEnabled && userHasAcceptedPreviousTos
   const [tosText, setTosText] = useState()
+  termsOfService.userContinuedUnderGracePeriod = false
```

Comment:
Instead of mutating `termsOfService` in this function, suggest creating a new variable to be passed to `authStore.update`.

```js
  const newTermsOfService = {
    ...termsOfService,
    userContinuedUnderGracePeriod: false,
  }
```

Reply:
I second this